File: solutions_1-10/Euler_1.py
## sum of all multiples of 3 and 5 below 1000? ##

import timeit


start = timeit.default_timer()

def euler_1():
    return sum([i for i in xrange(1, 1000) if i % 3 == 0 or i % 5 == 0])

print "Answer: %s" % euler_1()
stop = timeit.default_timer()
print "Time: %f" % (stop - start)

File: Java/java_solutions_1-10/Euler_1.java
/*
/   Project Euler problem #1 Multiples of 3 and 5
/
/   Find the sum of all the multiples of 3 or 5 below 1000
/
*/
import java.util.OptionalInt;
import java.util.stream.IntStream;

public class Euler_1
{
    public static int factors(int limit)
    {
        // Use OptionalInt over IntStream with a map-reduce like
        OptionalInt sum = IntStream.range(1, limit)
                                   .filter(x -> x % 3 == 0 ||
                                                x % 5 == 0)
                                   .reduce(Integer::sum);
        return sum.getAsInt();
    }

    public static void main(String[] args)
    {
        long start = System.nanoTime();

        int factor_sum = factors(1000);

        long stop = System.nanoTime();
        System.out.println("Answer: " + factor_sum);
        System.out.printf("Time: %.7f\n", ((float) stop - (float) start) / 10000000);
    }
}
